library/extractor/src/main/java/com/google/android/exoplayer2/metadata/id3/TextInformationFrame.java

```diff
@@ -19,52 +19,76 @@
 
 import android.os.Parcel;
 import android.os.Parcelable;
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.google.android.exoplayer2.MediaMetadata;
 import com.google.android.exoplayer2.util.Util;
+import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /** Text information ID3 frame. */
 public final class TextInformationFrame extends Id3Frame {
-
   @Nullable public final String description;
+
+  /** @deprecated Use {@code values[0]} instead. */
+  @Deprecated
   public final String value;
 
-  public TextInformationFrame(String id, @Nullable String description, String value) {
+  /** The text values of this frame. Will always have at least one element. */
+  @NonNull
+  public final List<String> values;
+
+  public TextInformationFrame(String id, @Nullable String description, @NonNull List<String> values) {
     super(id);
+    if (values.isEmpty()) {
+      throw new IllegalArgumentException("Text information frames must have at least one value");
+    }
+
     this.description = description;
-    this.value = value;
+    this.values = ImmutableList.copyOf(values);
+    this.value = this.values.get(0);
+  }
+
+  /** @deprecated Use {@code TextInformationFrame(String id, String description, String[] values} instead */
+  @Deprecated
+  public TextInformationFrame(String id, @Nullable String description, String value) {
+    this(id, description, Collections.singletonList(value) );
   }
 
   /* package */ TextInformationFrame(Parcel in) {
     super(castNonNull(in.readString()));
     description = in.readString();
-    value = castNonNull(in.readString());
+    values = ImmutableList.copyOf(in.createStringArray());
+    value = values.get(0);
   }
 
   @Override
   public void populateMediaMetadata(MediaMetadata.Builder builder) {
+    // Depending on the context this frame is in, we either take the first value of a multi-value
+    // frame because multiple values make no sense, or we join the values together with a comma
```

Comment:
It's been pointed out that these strings may end up directly in a UI (e.g. in a playback notification) and using a comma to separate a list of items is not appropriate for all locales.

We have an [`exo_item_list` resource definition](https://github.com/google/ExoPlayer/blob/r2.18.2/library/ui/src/main/res/values/strings.xml#L127) which gives a localised way to build up a list of items but it's currently defined in the UI module, so not available from here (the extractor module).

I understand you sent this PR because you need access to these multiple values. Do you need them to be automatically put in the `MediaMetadata`  object (i.e. handled in this method) or is it enough for them to be available from `TextInformationFrame` and your app can handle them from there? We're considering just keeping the existing behaviour of this method for now, due to the additional complexity of moving `exo_item_list` to e.g. the common module and also accessing it here (we'd need a `Context`). But if that makes this PR completely pointless for you then we should probably rethink that simplification :)

Reply:
It's enough to just be available in `TextInformationFrame` @icbaker. Feel free to revert it back to it's original behavior. What I might propose is a refactor of MediaMetadata itself to handle multiple values, but that's something that can be done on your end.